backend/experiment/rules/musical_preferences.py

```diff
@@ -170,11 +170,12 @@ def next_round(self, session, request_session=None):
             )
             actions = [feedback]
         elif n_songs == session.experiment.rounds + 1:
-            like_results = session.result_set.filter(question_key='like_song')
+            like_results = session.result_set.filter(question_key='like_song', section_id__isnull=False)
             known_songs = session.result_set.filter(
                 question_key='know_song', score=2).count()
             all_results = Result.objects.filter(
-                question_key='like_song'
+                question_key='like_song',
+                section_id__isnull=False
```

Comment:
The filter condition is really only necessary in this place (filtering over *all* results), as the current session is not going to use an obsolete playlist. That being said, that extra filter criterion probably won't slow things down so massively that it's a problem in other places.

Reply:
Is it per definition impossible for a session to have a result set with results without sections? If so, then I will remove it. I haven't seen the `result_set` / `{entity)_set` before. Is that a Django query pattern?